### DISCUSSION TITLE
Understand refPoints column

### DISCUSSION THREAD

**Discussion by orItach** [milaboratory/mixcr#1138] [Q&A] [answered]
Hi, i'm new to MiXCR so maybe it's very easy question.
I'm tryning to split the targetSequences to his parts (J,D,V) using refPoints column.
I can't find how to read the refPoints column.
I'm have succesfully run analyze command: 
mixcr analyze takara-human-tcr-V2-cdr3 --species hsa plate1_R1.fastq.gz plate1_R2.fastq.gz result
then i open the result.TRB.tsv and see some lines like 
refPoints                                      | targetSequences
:::::::::0:-7:7:12:-4:-2:18:21:-7:36:::  | TGTGCCAGCAGGGAGAATCTCTCCTACGAGCAGTACTTC
my first and only explantion is the first part is from 0 to 7 (TGTGCCA), with 7 deletion from ref seq,
the last part is from 21 to 36. 
But i can't understood the second part it's from 12 to 18 with 4 deletion and then 2 deletion? so why not simply write "12:-6:18"

**Comment by mizraelson** [comment, marked as answer]
Hi,
The rules for decoding refPoints column can be found [here](https://mixcr.com/mixcr/reference/mixcr-export/#default-anchor-point-positions).
Briefly each number defines a position of the reference anchor point on the target sequence.
Lets break down your case:
refPoints value - :::::::::0:-7:7:12:-4:-2:18:21:-7:36:::

1) The first number we see is 0 at the position 10 (we count positions between colons starting from 1, 1:2:3:4:5:6:7:8:9:10...
According to the documentation link above, 10th number means the beginning of the CDR3. Thus, the first nucleotide of the target sequence is the beginning of CDR3.
2) Next number is -7 at the position 11. If negative it is the number of deletions at the 3' end of V gene.
3) 7 at the position 12. 8th nucleotide of the target sequence is the first nucleotide after V gene. (7th nucleotide is the last V gene nucleotide)
4) 12 at the position 13. First nucleotide aligned to D reference.
5) -4 at the position 14. If negative - number of deletions on the 5' end of D segment.
6) -2 at the position 15. If negative - number of deletions on the 3' end of D segment.
Notice, we don't write -6 because then we cant tell how many nucleotides were trimmed from each side of D!

7) 18 at the position 16. 19th nucleotide is the first after D segment. (18th nucleotide is the last of the D segment)
8) 21 at the position 17. 22st nucleotide is the first aligned to J gene.
9) -7 at the position 18. Number of 3' J gene deletions.
10) 36 at the position 19. The end of CDR3.

Let me know if helps, feel free to ask any questions.

**Reply by orItach** [reply]
Thanks for the quick reply.
Just to make sure i understand correctly if i would like to split the targetSequences to his part (VDJ) all i need to do is read by index.
e.g. VSeq is TGCAGCG, seven nucleotide from 0 to 7, DSeq is CAGGGG six nucleotide from 12 to 18?
